### ISSUE TITLE
Server hangs when `Regex.matches` has a  `pattern` which matches the empty string 

### ISSUE THREAD

**Issue by pierce314159** [Bears-R-Us/arkouda#959]
Methods that use `Regex.matches` cause the server to hang when called with a `pattern` that matches the empty string (`''`, `'^'`, `'$'`,`'|'`, ...). This is due to https://github.com/chapel-lang/chapel/issues/18639

Reproducer:
```python
>>> strings = ak.array(['one'])
>>> strings.find_locations('')
```

The methods which use `matches` on current release include:
- `Strings.flatten(regex=True)`
- `Strings.find_locations`
- `Strings.findall`
- `Strings.peel(regex=True)`

Including functions introduced in PR #958:
- `Strings.search`
- `Strings.match`
- `Strings.fullmatch`
- `Strings.split`
- `Strings.sub/subn`
- `Match.find_matches`
- `Match.group`

I think the best course is to add a check for patterns that match empty strings
``` python
if re.search(pattern, ''):
    raise ValueError("patterns that match the empty string are not currently supported")
```

**Comment by pierce314159** [comment]
PR #962 provides a temporary fix for the server hanging issue, I think we should keep this issue open until the chapel changes make their way into arkouda so we remember to revisit

**Comment by ronawho** [comment]
@pierce314159 I believe this was fixed with Chapel 1.25.1, which is recommended as of https://github.com/Bears-R-Us/arkouda/pull/1027. Is that good enough to close this, or do you want to wait until 1.25.1 is the oldest version supported so you can remove the workaround?

**Comment by reuster986** [comment]
I'm going to close this, because the problem is resolved from the user's perspective. We can open a new issue to remove the temporary fix when we drop support for <1.25.1
